storage/innobase/log/log0recv.cc

```diff
@@ -1201,7 +1201,9 @@ inline void recv_sys_t::trim(const page_id_t page_id, lsn_t lsn)
   DBUG_VOID_RETURN;
 }
 
-inline dberr_t recv_sys_t::read(os_offset_t total_offset, span<byte> buf)
+inline
+__attribute__((warn_unused_result))
+dberr_t recv_sys_t::read(os_offset_t total_offset, span<byte> buf)
```

Comment:
I think that it is sufficient to add the attribute to the function declaration in `log0recv.h`.

If you want to add it here, please add it in a separate line on its own (before the `inline` keyword) and do not touch the existing line.